File: 392.is-subsequence.ts
/*
 * @lc app=leetcode id=392 lang=typescript
 *
 * [392] Is Subsequence
 */

// @lc code=start
function isSubsequence(s: string, t: string): boolean {
  let mainIndex = 0;

  for (let i = 0; i < s.length; i++) {
    let found = false;
    while (mainIndex < t.length) {
      if (s[i] == t[mainIndex]) {
        found = true;
        mainIndex++;
        break;
      } else {
        mainIndex++;
      }
    }
    if (!found) {
      return false;
    }
  }

  return true;
}
// @lc code=end

File: 392.is-subsequence.go
/*
 * @lc app=leetcode id=392 lang=golang
 *
 * [392] Is Subsequence
 */

package main

// @lc code=start
func isSubsequence(s string, t string) bool {
	for subI, mainI, found := 0, 0, false; subI < len(s); subI++ {
		found = false
		for mainI < len(t) {
			if s[subI] == t[mainI] {
				mainI++
				found = true
				break
			} else {
				mainI++
			}
		}
		if !found {
			return false
		}
	}

	return true
}

// @lc code=end
